src/net/sf/memoranda/ContactManager.java

```diff
@@ -9,372 +9,435 @@
+	 * the project contact list
+	 * 
+	 * @param project The Project for which the Document will be returned
+	 * @return nu.xom.Document object containing an XML representation of the
+	 *         project contact list
 	 */
 	public static Document toProjectDocument(Project project) {
-		Element rootElement = new Element("contactlist");
-		ArrayList<Contact> contacts = getContacts(project);
+		final Element rootElement = new Element("contactlist");
+		final ArrayList<Contact> contacts = getContacts(project);
+
 		contacts.forEach((contact) -> {
-			// Don't save any projects the contact is associated with since it's not needed.
-			Contact newContact = contact.copy();
+			// Don't save any projects the contact is associated with since it's
+			// not needed.
+			final Contact newContact = contact.copy();
 			newContact.clearIDs();
 			rootElement.appendChild(newContact.toElement());
 		});
+
 		return new Document(rootElement);
 	}
-	
-	
-	/** Adds a <code>ContactManagerListener</code> which is notified when a contact has
-	 * been added, removed, updated, or when a contact list has been imported
+
+	/**
+	 * Adds a <code>ContactManagerListener</code> which is notified when a
+	 * contact has been added, removed, updated, or when a contact list has been
+	 * imported.
+	 * 
 	 * @param listener The listener object to be added
-	 * @return Returns true if the listener could be added and false if it could not
+	 * @return Returns true if the listener could be added and false if it could
+	 *         not
 	 */
 	public static boolean addContactManagerListener(ContactManagerListener listener) {
 		return _listeners.add(listener);
 	}
-	
-	
-	/** Removes a <code>ContactManagerListener</code>
+
+	/**
+	 * Removes a <code>ContactManagerListener</code>.
+	 * 
 	 * @param listener The listener object to be removed
-	 * @return Returns true if the listener could be removed and false if it could not
+	 * @return Returns true if the listener could be removed and false if it
+	 *         could not
 	 */
 	public static boolean removeContactManagerListener(ContactManagerListener listener) {
 		return _listeners.remove(listener);
 	}
-	
-	
-	private ContactManager() {}
-	
-	
+
 	private static void _buildElements(Document doc) {
-		Element root = doc.getRootElement();
-		if(!root.getLocalName().equals("contactlist")) throw new NoSuchChildException("contactlist");
-		int size = root.getChildElements().size();
-		
-		// Master contacts hashmap size should be a minimum of 100 and increase in MASTERLF intervals
+		final Element root = doc.getRootElement();
+
+		if (!root.getLocalName().equals("contactlist")) {
+			throw new NoSuchChildException("contactlist");
+		}
+
+		// Master contacts hashmap size should be a minimum of 100 and increase
+		// in MASTERLF intervals
+		final int size = root.getChildElements().size();
 		int initHashSize = MASTERLF * ((size / MASTERLF) + 1);
-		if(initHashSize < 100) {
+
+		if (initHashSize < 100) {
 			initHashSize = 100;
```

Comment:
Lines 302 & 303: Use of literal values (CG:6). Recommend making 100 constant, such as MASTER_MIN.